### PR TITLE
Print trivia of SynMatchClause

### PR DESCRIPTION
**Pull request by HumbertoCortes**
The comments inside `with` block (of a try-with) are formatted. In the Clauses generation method ("genClause") a genTrivia call ("genTriviaMainNodesBeforeClausePipe") is inserted in the CodePrinter. The new method look for a trivia nodes with in TriviaMainNodes list.

Two unit test are added to test the correct behaviour.

### CODE REVIEW

**Comment by knocte** [comment]
@nojaf CI on 1st commit was broken because Humberto forgot to run fantomas on fantomas itself, but 2nd commit should have passed, however, the failure seems unrelated to this PR (something about paket...?).

File: src/Fantomas.Tests/TriviaTests.fs [hunk 1]
```diff
@@ -489,3 +490,65 @@ let ``number expression`` () =
     | [ { ContentItself = Some (Number (n))
           Type = TriviaNodeType.MainNode (SynExpr_Const) } ] -> n == "2.0m"
     | _ -> fail ()
+
+[<Test>]
+let ``line comment inside short `with` block (of a try-with), 1219`` () =
```
Comment (nojaf) [inline src/Fantomas.Tests/TriviaTests.fs:495]: Please move these tests to ControleStructureTests.fs

File: src/Fantomas/CodePrinter.fs [hunk 2]
```diff
@@ -4103,6 +4103,7 @@ and genClause astContext hasBar (Clause (p, e, eo)) =
         +> leaveNodeTokenByName arrowRange RARROW
 
     genTriviaBeforeClausePipe p.Range
+    +> genTriviaMainNodesBeforeClausePipe p.Range
```
Comment (nojaf) [inline src/Fantomas/CodePrinter.fs:4106]: You introduce a very specific helper here. I'd prefer that an existing helper is used to print the trivia.
`|> genTriviaFor SynMatchClause_Clause p.Range`.

However, this does not solve your indentation problem.
I would add an extra check if you have TriviaContentBefore the `SynMatchClause`.

https://github.com/fsprojects/fantomas/blob/ca42f0d7cf05144c5838bba725378e2a856211bb/src/Fantomas/CodePrinter.fs#L2040

Something like

```fsharp
            let hasCommentBeforeClause = true // TODO: add actual check in TriviaHelpers
            let genClause (astContext: ASTContext) (b: bool) (c: SynMatchClause) =
                ifElse
                    hasCommentBeforeClause
                    (indent +> genClause astContext b c +> unindent)
                    (genClause astContext b c)
```

And if that is the case wrap the genClause between indents.

File: src/Fantomas/Context.fs [hunk 3]
```diff
@@ -1448,6 +1448,15 @@ let internal genTriviaBeforeClausePipe (rangeOfClause: range) ctx =
         | None -> id
     <| ctx
 
+let internal genTriviaMainNodesBeforeClausePipe (rangeOfClause: range) ctx =
+    (Map.tryFindOrEmptyList SynMatchClause_Clause ctx.TriviaMainNodes)
+    |> List.tryFind (fun t -> RangeHelpers.rangeEq t.Range rangeOfClause)
+    |> fun trivia ->
+        match trivia with
+        | Some trivia -> indent +> (printContentBefore trivia)
```
Comment (nojaf) [inline src/Fantomas/Context.fs:1456]: `indent` without matching `unindent` is a very dangerous thing.
You need to add another test to verify that the indentation is correct again.
Something like:

```fsharp
[<Test>]
let ``foo `` () =
    formatSourceString false """
let outer =
    try
        //humberto
        TrySomething(someParam)
    with
        //comentario
        ex -> MakeSureToCleanup(someParam)

    let someBinding v = v + 2
"""  config
    |> prepend newline
    |> should equal """
let outer =
    try
        //humberto
        TrySomething(someParam)
    with
        //comentario
        ex -> MakeSureToCleanup(someParam)

    let someBinding v = v + 2
"""
```

But as mention above, I'd prefer to not have this helper function.

**Review by nojaf — CHANGES_REQUESTED** [review]
Hello @HumbertoCortes, thank you for this contribution.
Unfortunately, I'd like to see a different implementation.
Please let me know if you need further assistance.

File: src/Fantomas/Context.fs (hunk 3 above)
Comment (HumbertoCortes) [reply src/Fantomas/Context.fs:1456]: @nojaf Thanks so much for your comments. I will work in that direction!

**Review by nojaf — APPROVED** [review]
Hey @HumbertoCortes, looks good.
Thanks for the requested changes.
I've tweaked some stuff here and there to work with the full range of SynMatchClause.
Related https://github.com/dotnet/fsharp/issues/10877

**Comment by nojaf** [comment]
Fixes #1219

**Pull request merged**